omega/omega-connector/omega-connector-grpc/src/main/java/org/apache/servicecomb/saga/omega/connector/grpc/LoadBalancedClusterMessageSender.java

```diff
@@ -150,13 +153,14 @@ private MessageSender fastestSender() {
         .filter(entry -> entry.getValue() < Long.MAX_VALUE)
         .min(Comparator.comparingLong(Entry::getValue))
         .map(Entry::getKey)
-        .orElse((event -> {
+        .orElse(event -> {
           try {
-            availableMessageSenders.take().send(event);
-          } catch (InterruptedException e) {
+            return availableMessageSenders.take().send(event);
+          } catch (InterruptedException ignored) {
             Thread.currentThread().interrupt();
           }
-        }));
+          throw new OmegaException("Failed to send event " + event + " due to interruption");
```

Comment:
preferably to move this line to catch clause and pass interrupted exception to its constructor